libsolidity/parsing/Parser.cpp

```diff
@@ -370,6 +370,28 @@ std::pair<ContractKind, bool> Parser::parseContractKind()
 	return std::make_pair(kind, abstract);
 }
 
+ASTPointer<StorageLayoutSpecifier> Parser::parseStorageLayoutSpecifier()
+{
+	RecursionGuard recursionGuard(*this);
+	ASTNodeFactory nodeFactory(*this);
+	solAssert(*expectIdentifierToken() == "layout");
```

Comment:
Putting this in an assert makes all of it look like a sanity check but this is a part of the main logic here, because it has an side-effect - advancing the scanner. You're also dereferencing it without checking for null.

```suggestion
	ASTPointer<ASTString> layoutIdentifier = expectIdentifierToken();
	solAssert(layoutIdentifier && *layoutIdentifier == "layout");
```